src/components/LocalePicker.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import {ValueOf} from 'type-fest';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@styles/themes/useTheme';
+import useThemeStyles from '@styles/useThemeStyles';
+import * as App from '@userActions/App';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import Picker from './Picker';
+import type {PickerSize} from './Picker/types';
+
+type LocalePickerOnyxProps = {
+    /** Indicates which locale the user currently has selected */
+    preferredLocale: OnyxEntry<ValueOf<typeof CONST.LOCALES>>;
+};
+
+type LocalePickerProps = LocalePickerOnyxProps & {
+    /** Indicates size of a picker component and whether to render the label or not */
+    size?: PickerSize;
+};
+
+function LocalePicker({preferredLocale = CONST.LOCALES.DEFAULT, size = 'normal'}: LocalePickerProps) {
```

Comment:
I am curious if we can add consts for `size`? I see that `'normal'` is repeated a few times. I think it makes sense more, than type (`PickerSize`).